[sweet api][Kotlin] Dispatch async cpp calls through the native module queue

lukmccall opened the pull request:
# Why

a follow up to the https://github.com/expo/expo/pull/17020.

# How

- Created a single thread queue for dispatching the functions of the native module. 
- Dispatch CPP async call to the created queue

# Test Plan

- bare-expo ✅

lukmccall committed:
[sweet api][Kotlin] Dispatch async cpp calls through the native module queue

tsapeta commented on packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt:
```diff
@@ -32,6 +45,11 @@ class AppContext(
+  internal val moduleQueue = CoroutineScope(
+    newSingleThreadContext("ExpoModulesCoreQueue") +
+      SupervisorJob() +
+      CoroutineName("ExpoModuleCoreCoroutineQueue")
```
```suggestion
      CoroutineName("ExpoModulesCoreCoroutineQueue")
```

tsapeta commented on packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt:
```diff
@@ -32,6 +45,11 @@ class AppContext(
   }
   private val reactLifecycleDelegate = ReactLifecycleDelegate(this)
   private val jsiInterop = JSIInteropModuleRegistry(this)
+  internal val moduleQueue = CoroutineScope(
```
```suggestion
  internal val modulesQueue = CoroutineScope(
```
or
```suggestion
  internal val modulesCoroutineScope = CoroutineScope(
```

lukmccall committed:
[core][Android] Fix unit tests & add setup for UI tests (#17433)

# Why

Fixes unit tests in `e-m-c`.
Adds support for Android tests.

# How

Ensures that so files will be loaded in instrumental tests and postpones the creation of the `JSIInteropModuleRegistry` to not load static libraries in unit tests. 

# Test Plan

- bare-expo ✅
- unit tests ✅

lukmccall committed:
[skip ci] Apply requested changes

Co-authored-by: Tomasz Sapeta <tomasz.sapeta@swmansion.com>
